examples/gadgets/simple/trace_dns/main.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	igjson "github.com/inspektor-gadget/inspektor-gadget/pkg/datasource/formatters/json"
+	gadgetcontext "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-context"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager"
+	ocihandler "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/oci-handler"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/simple"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/socketenricher"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+)
+
+func do() error {
+	const opPriority = 50000
+	myOperator := simple.New("myHandler", simple.OnInit(func(gadgetCtx operators.GadgetContext) error {
+		for _, d := range gadgetCtx.GetDataSources() {
+			jsonEncoder, _ := igjson.New(d)
+
+			d.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				jsonOutput := jsonEncoder.Marshal(data)
+				fmt.Printf("%s\n", jsonOutput)
+				return nil
+			}, opPriority)
+		}
+		return nil
+	}))
+
+	// There are some gadgets, like trace_dns, trace_sni, trace_network that
+	// require the local manager (or kube manager) operators to work.
+	// TODO: How to mark them and why is it a hard requirement?
+
+	// Create the local manager operator
+	localManagerOp := &localmanager.LocalManager{}
+	localmanagerParams := localManagerOp.GlobalParamDescs().ToParams()
+	localmanagerParams.Get(localmanager.Runtimes).Set("docker")
+	localManagerOp.Init(localmanagerParams)
```

Comment:
Mmmmmm I'm not sure we should manually init operators. IMO, we should pass the global params to gadget context, and it should init the operator for us (if operator is needed). BTW, what you're doing here is currently overwritten here, isn't it?

https://github.com/inspektor-gadget/inspektor-gadget/blob/e375eb658ecddedaa03471be0f34d7d66be54abb/pkg/gadget-context/run.go#L44-L51

On the other hand, how to manger the global params used for initialization? Should we use the ones coming from a config file or so and overwrite them if something is pass here?

Reply:
We're currently not doing it correctly. IMHO, the caller should always make sure operators are initialized properly instead of the `gadgetCtx` (which should only do the instance initialization).

Basically all our entry points (ig/main.go, kubectl-gadget/main.go, gadgetctl/main.go and gadgettracermgr) should initialize all operators globally from a config file (or params or env, basically the cobra/viper combo). It could be done by a wrapper so that it works the same way for all of them.

Reply:
I totally second @flyth, operators should be initialized outside the gadget-context (it's possible to use the same operator for running multiple gadgets). I'll give it a try later on (otherwise I'll open an issue).

Reply:
BTW, what about the simple operator? Should we call also `myOperator.Init(nil)`? Or we want to skip it as we know it does nothing?

Reply:
I think we can skip it.

Reply:
Opened #2891